ethereum/swap_factory_test.go

```diff
@@ -44,18 +46,71 @@ func testNewSwap(t *testing.T, asset ethcommon.Address) {
 	require.NotEqual(t, ethcommon.Address{}, address)
 	require.NotNil(t, tx)
 	require.NotNil(t, contract)
+
 	receipt, err := block.WaitForReceipt(context.Background(), conn, tx.Hash())
 	require.NoError(t, err)
 	t.Logf("gas cost to deploy SwapFactory.sol: %d", receipt.GasUsed)
 
-	nonce := big.NewInt(0)
-	tx, err = contract.NewSwap(auth, [32]byte{}, [32]byte{},
-		ethcommon.Address{}, defaultTimeoutDuration, asset, big.NewInt(0), nonce)
+	owner := auth.From
+	claimer := common.EthereumPrivateKeyToAddress(tests.GetMakerTestKey(t))
+
+	var pubKeyClaim, pubKeyRefund [32]byte
+	_, err = rand.Read(pubKeyClaim[:])
 	require.NoError(t, err)
-	receipt, err = block.WaitForReceipt(context.Background(), conn, tx.Hash())
+	_, err = rand.Read(pubKeyRefund[:])
+	require.NoError(t, err)
+
+	nonce, err := rand.Prime(rand.Reader, 256)
+	require.NoError(t, err)
+	value, err := rand.Prime(rand.Reader, 53) // (2^54 - 1) / 10^18 ~= .018 ETH
+	require.NoError(t, err)
+
+	isEthAsset := asset == ethAssetAddress
+
+	if isEthAsset {
+		auth.Value = value
+	} else {
+		value = big.NewInt(0)
```

Comment:
The original test was always calling `newSwap` with a value of zero. To get this test, which is now testing our off-net calculation of the SwapID, working with ERC20 tokens, just setting the value back to zero, the way it previously was, removed the transaction revert. I didn't dig though the solidity, but I'm going to guess it failed because the message sender didn't have the tokens to lock.